tests/unit/amo/components/TestAddonReviewCard.js

```diff
@@ -300,228 +363,215 @@ describe(__filename, () => {
+      cancelDeleteAddonReview({
+        reviewId: review.id,
+      }),
     );
-    sinon.assert.called(fakeEvent.preventDefault);
   });
 
-  it('dispatches deleteReview when a user deletes a review', () => {
+  it('dispatches deleteReview and displayes a deleting message when a user deletes a review', () => {
     const review = signInAndDispatchSavedReview();
-    store.dispatch(beginDeleteAddonReview({ reviewId: review.id }));
-    const dispatchSpy = sinon.spy(store, 'dispatch');
-    const root = render({ review });
-    const { errorHandler } = root.instance().props;
+    const dispatch = jest.spyOn(store, 'dispatch');
+    render({ review });
 
-    const dialog = root.find('.AddonReviewCard-confirmDeleteDialog');
-    const deleteEvent = createFakeEvent();
+    clickDeleteReview();
 
-    // This emulates a user clicking the delete button and confirming.
-    const onDelete = dialog.prop('onConfirm');
-    onDelete(deleteEvent);
+    const button = screen.getByRole('button', { name: 'Delete' });
+    const clickEvent = createEvent.click(button);
+    const preventDefaultWatcher = jest.spyOn(clickEvent, 'preventDefault');
 
-    sinon.assert.calledOnce(deleteEvent.preventDefault);
-    sinon.assert.calledWith(
-      dispatchSpy,
-      deleteAddonReview({
-        addonId: review.reviewAddon.id,
-        errorHandlerId: errorHandler.id,
-        reviewId: review.id,
-      }),
-    );
-  });
+    fireEvent(button, clickEvent);
 
-  it('renders a deleting message when a user deletes a review', () => {
-    const review = signInAndDispatchSavedReview();
-    store.dispatch(
+    expect(preventDefaultWatcher).toHaveBeenCalled();
+    expect(dispatch).toHaveBeenCalledWith(
       deleteAddonReview({
-        addonId: fakeAddon.id,
-        errorHandlerId: createStubErrorHandler().id,
+        addonId: review.reviewAddon.id,
+        errorHandlerId: getDefaultErrorHandlerId(review.id),
         reviewId: review.id,
       }),
     );
 
-    const root = render({ review });
-
-    const controls = renderControls(root);
-    expect(controls.find('.AddonReviewCard-deleting')).toHaveLength(1);
-    expect(controls.find('.AddonReviewCard-delete')).toHaveLength(0);
+    expect(
+      screen.queryByRole('button', { name: 'Delete review' }),
+    ).not.toBeInTheDocument();
+    expect(screen.getByText('Deleting…')).toBeInTheDocument();
   });
 
   it('renders a delete link when an error occurs when deleting a review', () => {
     const review = signInAndDispatchSavedReview();
-    store.dispatch(
-      deleteAddonReview({
-        addonId: fakeAddon.id,
-        errorHandlerId: createStubErrorHandler().id,
-        reviewId: review.id,
-      }),
-    );
+    render({ review });
 
-    const errorHandler = new ErrorHandler({
-      id: 'some-id',
-      dispatch: store.dispatch,
-    });
-    errorHandler.handle(new Error('some unexpected error'));
+    clickDeleteReview();
+    userEvent.click(screen.getByRole('button', { name: 'Delete' }));
```

Comment:
where do we set up that the error occurs?  (The only reference to errors - `createFailedErrorHandler` - is after the click has taken place)